webapp/javascript/hooks/populateLeftRightQuery.hook.ts

```diff
@@ -1,18 +1,24 @@
 import { useEffect } from 'react';
 import { actions, selectQueries } from '@webapp/redux/reducers/continuous';
 import { useAppDispatch, useAppSelector } from '@webapp/redux/hooks';
+import { queryToAppName, Query } from '@webapp/models/query';
 
-function isQueriesHasSameApp(queries: string[]): boolean {
-  const appName = queries[0].split('{')[0];
+function isQueriesHasSameApp(queries: Query[]): boolean {
+  const appName = queryToAppName(queries[0]);
+  if (appName.isNothing) {
+    return false;
+  }
 
-  return queries.every((query) => query.match(appName));
+  return queries.every((query) => query.match(appName.value));
```

Comment:
Would this be better?
```suggestion
  return queries.every((query) => query.startsWith(appName.value));
```